main.go

```diff
@@ -155,6 +153,11 @@ func mustGetProfilePath() string {
 func main() {
 	app := registerApp()
 	app.Before = func(c *cli.Context) error {
+		// Check if running as 'root' is allowed.
+		if globalRootDisallow {
```

Comment:
here is where you would check and exit directly. Another alternative is to remove isContainerized and root check. Leave it up to the user to take care. More and more Unix privileges are becoming irrelevant.